enhancements/cnf-update-reference-config-hub-side-templ.md

```diff
@@ -0,0 +1,173 @@
+---
+title: Update reference configuration to use hub side templating
+authors:
+  - "@irinamihai"
+reviewers:
+  - "@imiller0"
+  - "@serngawy"
+approvers:
+  - "@imiller0"
+
+creation-date: 2023-08-17
+last-updated: 2023-08-21
+---
+
+
+# Update reference configuration to use hub side templating
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Documentation is updated in the upstream
+- [ ] Documentation is updated in the downstream
+
+
+## Summary
+
+* The current ZTP reference configuration includes examples of common, group and site policies.
+* As per their names
+  * site policies contain configuration specific to one specific cluster
+  * group policies configuration common to a group of clusters
+  * common policies contain configuration common to all sites.
+* The group and site policies generally have an identical structure, with the cluster specific configuration being different.
+* To reduce the number of policies one has to maintain and keep track of for a certain deployment, we can make use of a general purpose policy and have its per-cluster values substituted into a hub side template.
+
+
+## Motivation
+
+* With large fleets of clusters, it can quickly become hard to maintain per-site configurations.
+
+* Offer users an example for how to reduce the number of policies on their hub clusters. This would make the cluster’s deployment and maintenance process a lot less cumbersome.
+
+* Customers are already making use of hub side templating, so we should cover this scenario in our reference configuration.
+
+
+### Goals
+
+* Update the existing reference configuration with the following:
+  * hub side template examples and references to its documentation
+  * PolicyGenTemplates that use templates instead of hard-coded configuration
+  * configMaps that contain the configuration needed for the used PGTs
+* The user can include the desired configuration data in SiteConfig and SiteConfig Generator will create the needed configMaps
+* Explain if and how the ZTP workflow is impacted by switching to using hub side templates
+
+### Non-Goals
+* hub side templating for SiteConfig
```

Comment:
I was wondering about we are injecting site-date in the siteconfig, but hub-side templating targeting only policyGenTemplate but not sitefig itself?

Reply:
The target for this enhancement is just PGT. I think the reason is because we would need more pieces to make sure we have all the data necessary for the templates in SiteConfig. Adding the site-specific configuration under SiteConfig assures PGT will have all the needed data at the time of the configuration.

@imiller0, kindly correct me and/or add on the reason, please. Thank you!

Reply:
From my understanding, hubside-templating feature is coming from acm and it is applicable only to policies which we build using PGT. So, we can't have templating for siteconfig actually.

Reply:
That is correct, the templating only applies to policies. In our methodology the policies are defined in the PGT and will now contain templates. Those templates will need data that is per-site (or per-group in some cases). From a customer perspective the SiteConfig represents the "cluster specific data", ie there is one SiteConfig per cluster and it has all the attributes that are unique to that cluster. Today we use that to drive the installation flow, but with this (relatively small) change the SiteConfig also becomes the source for the per-site template data. When customers are building automation around our solution they now have a single CR that they are maintaining with all site data (some for installation, some for configuration). This is easier to maintain and automate against than multiple CRs, or CRs shared between sites.